broker/fragment/store_azure.go

```diff
@@ -167,6 +191,24 @@ func (a *azureBackend) Remove(ctx context.Context, fragment pb.Fragment) error {
 	return err
 }
 
+// https://gist.github.com/ItalyPaleAle/ec6498bfa81a96f9ca27a2da6f60a770
+func GetAzureStorageCredential(coreCredential azcore.TokenCredential) (azblob.TokenCredential, error) {
+	var tokenRefresher azblob.TokenRefresher
+	tokenRefresher = func(credential azblob.TokenCredential) time.Duration {
+		accessToken, err := coreCredential.GetToken(context.Background(), policy.TokenRequestOptions{Scopes: []string{"https://storage.azure.com/.default"}})
+		if err != nil {
+			panic(err)
+		}
+		credential.SetToken(accessToken.Token)
+
+		exp := accessToken.ExpiresOn.Sub(time.Now().Add(2 * time.Minute))
```

Comment:
where is two minutes coming from?

Reply:
This is a padding window to make sure we refresh the access token before it expires. 2 minutes seems reasonable, [given that](https://learn.microsoft.com/en-us/azure/active-directory/develop/configurable-token-lifetimes#access-tokens): 

> The default lifetime of an access token is variable. When issued, an access token's default lifetime is assigned a random value ranging between 60-90 minutes (75 minutes on average)

Reply:
Okay, that needs a comment.

Reply:
this particular line is moot because it's in that function that I should have removed, but I will add a comment over [here](https://github.com/gazette/core/pull/341/files#diff-b58442da8fead03ddc65f71858da8439f9f4b987fc32da9b453d9fba6a066c88R263) to the same effect (expiration buffer)